<table align="center" width=100%>
    <tr>
        <td width="20%">
            <img src="New_GL_Logo.png">
        </td>
        <td>
            <div align="center">
                <font color="#21618C" size=20px>
                    <b> Inclass-Lab  <br>(Day 3)
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

## Table of Content

1. **[Conditional Statements](#cond)**
2. **[Python Flow Control](#flow_control)**
3. **[List Comprehension](#lc)**



<a id="cond"> </a>
## 1. Conditional Statements

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>1. Write a program to check whether a given number is multiple of 7 (use if statement)</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [2]:
num=int(input("enter a number"))
if num%7==0:
    print("Multiple of 7")
else:
    print("not  a multiple of 7")

enter a number7
Mult of 7


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>2. Write a code to print the maximum of two numbers (use if-else)</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [3]:
a=int(input("enter a number"))
b=int(input("enter a number"))
if a>b:
    print("a is max")
else:
    print("b is max")

enter a number4
enter a number8
b is max


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>3. Write a program to check whether a number is positive, negative or zero (use if-elif) </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [4]:
num=int(input("enter a number"))
if num>0:
    print("no. is positive")
elif num<0:
    print("no.is negative")
else :
    print("no.is zero")
    

enter a number6
no. is positive


### Let's begin with some hands-on practice exercises

<a id= "flow_control"> </a>
## 2. Python Flow Control

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>4. Write a code to print table of 5 using assignment operand +=.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [31]:
n=5
for i in range(1,11):
    print(n)
    n+=5
    
    
    
    
    

5
10
15
20
25
30
35
40
45
50


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>5. Write a program for generating a fibonacci series starting with 0 and 1 for the next 10 values using a while loop. </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [27]:
a=0  #first no.
b=1  #sec no.
i=0
print(a)
print(b)
while i<11:
    c=a+b
    print(c)
    a=b
    b=c
    i+=1
    
    
    

0
1
1
2
3
5
8
13
21
34
55
89
144


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>6. Write a program to check whether a number is greater than the other using if statement (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [3]:
A=int(input("enter the 1st no."))
B=int(input("enter the 2nd no. "))
if A>B:
    print("A is greater")
else:
    print("B is greater")

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>7. Write a code to check whether a number is divisible by 7 or not (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [4]:
num=int(input("enter a number"))
if num%7==0:
    print("divisible by 7")
else:
    print("not divisible by 7")

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>8. Write a code to find factorial of a number (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [11]:
num=int(input("enter a number"))
f=1
if num==0:
    print("1")
elif num<0:
    print("invalid")
else:    
    for i in range(1,num+1):
        f=f*i
print(f)    

enter a number0
1


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>9. Write a program to check whether a number is prime or not (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [55]:
num=int(input("enter the no."))

for i in range(2,num):
    if num%i==0:
        print("not prime no.")
        break
else:
    print("prime")

enter the no.6
not prime no.


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>10. Write a program to check whether two numbers are amicable or not (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [64]:
num1=int(input("enter the no."))
l1=[]

for i in range(1,num1):
    if num1%i==0:
        l1.append(i)
num2=int(input("enter the no."))
l2=[]

for i in range(1,num2):
    if num2%i==0:
        l2.append(i)
        
        
if sum(l1)==num2 and sum(l2)==num1:
    print("A")
else:
    print("not A")




enter the no.220
enter the no.284


TypeError: 'int' object is not callable

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>11. Reverse string using a for loop (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [65]:
str1=str(input("enter a string"))
for i in str1[::-1]:
    print(i,end="")
   
    

enter a stringhello
olleh

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>12. Write a code to find the average of given numbers (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [70]:

numbers=(input("enter numbers "))
s=0
for i in numbers.split():
    s+=int(i)
print(s/len(numbers.split()))








enter numbers 6 7 8
7.0


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>13. Write a program to find the area of a circle for a given radius (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [14]:
rad=int(input("enter radius"))
area_cir=3.14*rad*rad
print(area_cir)

enter radius4
50.24


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>14. Write a code to find the simple interest (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [13]:
p=int(input("enter principal"))
r=int(input("enter rate of interest"))
t=int(input("time period"))
SI=(p*r*t)/100
print(SI)

enter principal1000
enter rate of interest45
time period3
1350.0


<a id= "lc"> </a>
## 3. List Comprehension

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>15. Use list comprehension to obtain the squre root of first 10 natural numbers.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [1]:
l1=[ i**0.5    for i in range(1,11)  ]
print(l1)

[1.0, 1.4142135623730951, 1.7320508075688772, 2.0, 2.23606797749979, 2.449489742783178, 2.6457513110645907, 2.8284271247461903, 3.0, 3.1622776601683795]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>16. Use list comprehension to find the used vowels in a given sentence and/or word (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [13]:
word=str(input("enter a word or sentence "))
l=[ i   for i in word if i if i in "aeiouAEIOU" ]
l


enter a word or sentence Energy


['E', 'e']

In [ ]:
l={ i   for i in word if i if i in "aeiouAEIOU" }
#set will give unique val

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>17. Create a dictionary and access it values using a condition on its key. The data is given below. Let the condidtion on the key is that it should be a even number.</b>
                </font>
            </div>
       </td>
    </tr>
</table>


<table>
<tr>
    <th>Key</th>
    <td>1</td>
    <td>2</td>
    <td>3</td>
    <td>4</td> 
    <td>5</td> 
</tr>
<tr>
    <th>Name</th>
    <td>Aman</td>
    <td>Mohit</td>
    <td>Guari</td>
    <td>Imran</td>
    <td>Roma</td
     </tr>
<tr>
    <th>Marks</th>
    <td>24</td>
    <td>25</td>
    <td>26</td>
    <td>24</td>
    <td>27</td>
   </tr>
</table>

In [14]:
d1={1:["aman",24],2:["mohit",25],3:["guari",26],4:["mran",24],5:["roma",27]}
d1

{1: ['aman', 24],
 2: ['mohit', 25],
 3: ['guari', 26],
 4: ['mran', 24],
 5: ['roma', 27]}

In [18]:
d2={i:d1[i] for i in d1.keys() if i%2==0}
d2

{2: ['mohit', 25], 4: ['mran', 24]}

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>18. Use list comprehension to find even and odd numbers from first 20 whole numbers.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [12]:
l2=[i  for i in range(0,21) if i%2==0 ]
l3=[i  for i in range(0,21) if i%2!=0 ]
print(l2,l3)

[0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20] [1, 3, 5, 7, 9, 11, 13, 15, 17, 19]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>19. Use list comprehension to print numbers divisible  by 2 and 3 in between 1 and 100. </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [22]:
l1=[ i   for i in range(1,101) if i%6==0]
print(l1)

[6, 12, 18, 24, 30, 36, 42, 48, 54, 60, 66, 72, 78, 84, 90, 96]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>20. Use list comprehension to create a dictionary such that its keys are numbers from 1 to 10 and values are the corresponding to that key are its cube.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [24]:
l1={i:i**3   for i in range(1,11)}
print(l1)

{1: 1, 2: 8, 3: 27, 4: 64, 5: 125, 6: 216, 7: 343, 8: 512, 9: 729, 10: 1000}


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>21. Use list comprehension to extract numbers from a string </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [26]:
str="hello123"
l1=[ i for i in str if i.isdigit()]
l1

['1', '2', '3']

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>22. Use list comprehension to print table of 11 to 20 as shown below.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

<table>
<tr>
    <td>11</td><td>22</td><td>33</td><td>44</td><td>55</td><td>66</td><td>77</td><td>88</td><td>99</td><td>110</td> 
</tr>
    
<tr>
    <td>12</td><td>24</td><td>36</td><td>48</td><td>60</td><td>72</td><td>84</td><td>96</td><td>108</td><td>120</td> 
</tr>

<tr>
    <td>13</td><td>36</td><td>39</td><td>52</td><td>65</td><td>78</td><td>91</td><td>104</td><td>117</td><td>130</td> 
</tr> 

<tr>
     <td>14</td><td>28</td><td>42</td><td>56</td><td>70</td> <td>84</td><td>98</td><td>112</td><td>126</td><td>140</td> 
</tr>

<tr>
     <td>15</td><td>30</td><td>45</td><td>60</td><td>75</td><td>90</td><td>105</td><td>120</td><td>135</td><td>150</td> 
</tr>

<tr>
     <td>16</td><td>32</td><td>48</td><td>64</td><td>80</td><td>96</td><td>112</td><td>128</td><td>144</td><td>160</td> 
</tr> 

<tr>
     <td>17</td><td>34</td><td>51</td><td>68</td><td>85</td><td>102</td><td>119</td><td>136</td><td>153</td><td>170</td> 
</tr> 

<tr>
     <td>18</td><td>36</td><td>54</td><td>72</td><td>90</td> <td>108</td><td>126</td><td>144</td><td>162</td><td>180</td> 
</tr>

<tr>
 <td>19</td><td>38</td><td>57</td><td>76</td><td>95</td><td>114</td><td>133</td><td>152</td><td>171</td><td>190</td> 
</tr>

<tr>
     <td>20</td><td>40</td><td>60</td><td>80</td><td>100</td><td>120</td><td>140</td><td>160</td><td>180</td><td>200</td> 
<tr>    
    
</table>

In [37]:
l1=[ [i*j for i in range(1,11)] for j in range(11,21) ]
l1

[[11, 22, 33, 44, 55, 66, 77, 88, 99, 110],
 [12, 24, 36, 48, 60, 72, 84, 96, 108, 120],
 [13, 26, 39, 52, 65, 78, 91, 104, 117, 130],
 [14, 28, 42, 56, 70, 84, 98, 112, 126, 140],
 [15, 30, 45, 60, 75, 90, 105, 120, 135, 150],
 [16, 32, 48, 64, 80, 96, 112, 128, 144, 160],
 [17, 34, 51, 68, 85, 102, 119, 136, 153, 170],
 [18, 36, 54, 72, 90, 108, 126, 144, 162, 180],
 [19, 38, 57, 76, 95, 114, 133, 152, 171, 190],
 [20, 40, 60, 80, 100, 120, 140, 160, 180, 200]]

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>23. Use list comprehension to find transpose of the matrix given below.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

<table>
    <tr>
        <td>1</td>
        <td>2</td>
    </tr>
    <tr>
        <td>3</td>
        <td>4</td>
    </tr>
    <tr>
        <td>5</td>
        <td>6</td>
    </tr>
    <tr>
        <td>7</td>
        <td>8</td>
    </tr>
    </table>

In [33]:
m=[
    [1,2],
    [3,4],
    [5,6],
    [7,8]
]
[ [ row[i] for row in m ] for i in range (2)]

[[1, 3, 5, 7], [2, 4, 6, 8]]